### PR TITLE
Feat: Graceful Handling of First-touch API Call (App List)

### LINKED ISSUE #11: [Feature]: Handle First-touch API Failure Gracefully
### Guidelines

- [X] I agree to follow this project's Contributing Guidelines.

### Description

As per #10 and the discussion on the thread, we've noticed that several scenarios may break the app list. One of these is Posit Connect not sending the correct JSON. We can handle this gracefully by showing an appropriate message and updating documentation.

### Problem

#10 has a good discussion about the problem. The gist is that the API's `/content` endpoint does not send the correct data if there is an OAuth layer in between.

### Proposed Solution

We can add proper messaging and handling of several scenarios.

### Alternatives Considered

No particular alternatives are relevant in this case.

### PR DESCRIPTION
**Pull request by DeepanshKhurana**
*Have you read the [Contributing Guidelines](https://github.com/Appsilon/.github/blob/main/CONTRIBUTING.md)?* ✅ 

Closes #11 
Related to #10 

## Description
- Handle the case where the API works but has no apps (`reactableLang`'s `noData` for the app table)
- Handle the case where the API itself fails by adding a new, more relevant empty state
- Minor fixes for CSS and strings
- Update README.md with FAQs

The GIF below illustrates the changes/cases.

Screen Recording 2024-06-11 at 4 04 47 PM

## Definition of Done
- [x] The change is thoroughly documented.
- [x] The CI passes (`R CMD check`, linter, unit tests, spelling).
- [x] Any generated files have been updated (e.g. `.Rd` files with `roxygen2::roxygenise()`)

### CODE REVIEW

File: app/logic/empty_state_utils.R [hunk 1]
```diff
@@ -0,0 +1,32 @@
+box::use(
+  shiny[
+    div,
+    img,
+    p,
+    renderUI
```
Comment (Gotfrid) [inline app/logic/empty_state_utils.R:6]: nitpick: prefer trailing comma

File: app/logic/empty_state_utils.R [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+#' @description Function to generate an empty state UI
+#' @param text Text to display in the empty state
+#' @param image_path Path to the image to display in the empty state
+#' @export
+generate_empty_state_ui <- function(
+  text = "Select an application and a job to view logs",
+  image_path = "static/illustrations/empty_state.svg"
+) {
+  renderUI({
```
Comment (Gotfrid) [inline app/logic/empty_state_utils.R]: suggestion: i think it's better to make a function that is not shiny-specific, i.e. remove `renderUI` from it. This way you can control what to do with the output of this function - use it in the server, or in the ui function.

File: app/main.R [hunk 3]
```diff
@@ -101,12 +96,16 @@ server <- function(id) {
...
           state
         )
+
       } else {
+
         removeUI(ns("job_list_pane"))
+
       }
-    }, ignoreInit = TRUE, ignoreNULL = TRUE)
+    }, ignoreInit = FALSE, ignoreNULL = FALSE)
```
Comment (Gotfrid) [inline app/main.R]: nitpick: `ignoreInit = FALSE` is the default value, no need to specify it.

File: app/main.R [hunk 4]
```diff
@@ -118,21 +117,22 @@ server <- function(id) {
...
-              class = "empty-state-text",
-              "Select an application and a job to view logs"
-            ),
-            img(
-              src = "static/empty_state.svg",
-              class = "empty-state-image",
-              alt = "Select an application and a job to view logs"
-            )
+
+        if (class(app_list()) != "data.frame") {
```
Comment (Gotfrid) [inline app/main.R]: suggestion: prefer `inherits(app_list(), "data.frame")
example:
```r
x <- dplyr::as_tibble(mtcars)
class(x) # "tbl_df"     "tbl"        "data.frame"
class(x) != "data.frame" TRUE  TRUE FALSE
inherits(x, "data.frame") # TRUE
```

**Review by Gotfrid — APPROVED** [review]
Nice work, thank you for your rapid reaction to an opened issue 💯 
Approved with a few non-critical comments.

Another comment (out of scope of this PR) I have is related to a pattern that I see in main.R and which I find weird:
```r
observeEvent(..., { renderUI(mod$ui()); mod$server() }
```

Would it be cleaner if you create ui+server for a particular app section only once, and inside a corresponding server you decide what to render by passing a reactive key, e.g. `state$selected_app()$guid`?

**Comment by DeepanshKhurana** [comment]
Thank you @Gotfrid for all the suggestions. I've made changes and merged them. For the comment about the reactivity flow, I can't think of the original reason for taking that route in implementation. I agree with you that it is a bit awkward and can be refactored. Documented this as an enhancement issue (#13)

**Pull request merged**